src/ddprof_worker.cc

```diff
@@ -631,22 +686,24 @@ void ddprof_pr_lost(DDProfContext *ctx, const perf_event_lost *lost,
   ctx->worker_ctx.lost_events_per_watcher[watcher_pos] += lost->lost;
 }
 
-void ddprof_pr_comm(DDProfContext *ctx, const perf_event_comm *comm,
-                    int watcher_pos) {
+DDRes ddprof_pr_comm(DDProfContext *ctx, const perf_event_comm *comm,
+                     int watcher_pos) {
   // Change in process name (assuming exec) : clear all associated dso
   if (comm->header.misc & PERF_RECORD_MISC_COMM_EXEC) {
     LG_DBG("<%d>(COMM)%d -> %s", watcher_pos, comm->pid, comm->comm);
-    unwind_pid_free(ctx->worker_ctx.us, comm->pid);
+    DDRES_CHECK_FWD(worker_pid_free(ctx, comm->pid));
   }
+  return ddres_init();
 }
 
-void ddprof_pr_fork(DDProfContext *ctx, const perf_event_fork *frk,
-                    int watcher_pos) {
+DDRes ddprof_pr_fork(DDProfContext *ctx, const perf_event_fork *frk,
+                     int watcher_pos) {
   LG_DBG("<%d>(FORK)%d -> %d/%d", watcher_pos, frk->ppid, frk->pid, frk->tid);
   if (frk->ppid != frk->pid) {
     // Clear everything and populate at next error or with coming samples
-    unwind_pid_free(ctx->worker_ctx.us, frk->pid);
```

Comment:
Why do we need to call `worker_pid_free` on newly created process upon fork ?

Reply:
in case PID was already used by something else ? (yes, very unlikely)

Reply:
I'm guessing this is probably bad actually, we might be forcing backpopulate mechanisms in case things arrive out of order.